js/src/vector/flat.ts

```diff
@@ -183,6 +183,15 @@ export class DateDayView extends PrimitiveView<Date_> {
 
 export class DateMillisecondView extends FixedSizeView<Date_> {
     public toArray() { return [...this]; }
+    public indexOf(search: Date) {
+        let index = 0;
+        for (let value of this) {
+            if (value.getTime() === search.getTime()) { return index; }
```

Comment:
@TheNeuralBit One way to optimize this search is to store the result of `search.valueOf()` at the start and apply a view that doesn't convert the raw bytes to dates. We can move this function to be an override of `indexOf()` on the DateVector itself, then it can look like this:
```ts
class DateVector {
    indexOf(search: Date) {
        return this.asEpochMilliseconds().indexOf(search.valueOf());
    }
}
```

Reply:
of course! good call, adding the change now